pageserver/src/tenant/size.rs

```diff
@@ -270,6 +265,8 @@ pub(super) async fn gather_inputs(
                 segment: Segment {
                     parent: Some(parent),
                     lsn: lsn.0,
+                    parent_lsn: Some(segments[parent].segment.lsn),
+                    wal_from_parent: Some(lsn.0 - segments[parent].segment.lsn),
```

Comment:
Not sure if a good thing to do, but to note: I vaguely remember there were already some overflow-related issues with size calculation, so might it be better to avoid potential overflows explicitly, with `u64::checked_sub` or similar?